dags/hindawi/hindawi_api_client.py

```diff
@@ -10,12 +10,12 @@ def __init__(
         base_url=None,
         path_segments=None,
     ):
-        self.base_url = base_url or getenv(
-            "HINDAWI_API_BASE_URL", "https://www.hindawi.com"
-        )
-        self.path_segments = path_segments or ["oai-pmh", "oai.aspx"]
+        self.base_url = base_url
+        self.path_segments = path_segments
 
     def get_articles_metadata(self, parameters, doi=None):
+        self.base_url = getenv("HINDAWI_API_BASE_URL", "https://www.hindawi.com")
+        self.path_segments = ["oai-pmh", "oai.aspx"]
```

Comment:
the `path_segments` are set in the init, why we want to override them here?

Reply:
because for getting files, path segments are different 
url for getting metadata: `https://www.hindawi.com/oai-pmh/oai.aspx?verb=listrecords&set=HINDAWI.AHEP&metadataprefix=marc21&from=...`
url for getting files: `http://downloads.hindawi.com/journals/ahep/...`